prototypes/stats/frontend/app/components/StatsApp.tsx

```diff
@@ -16,22 +19,32 @@ import { useState } from "react";
 export default function StatsApp() {
   const { getRootProps, getInputProps, isDragActive } = useDropzone({ onDrop });
   const [data, setData] = useState<StatData | null>(null);
+  const [loaded, setLoaded] = useState(false);
+  const [tileData, setTileData] = useState<TileData | null>(null);
 
   async function onDrop(acceptedFiles: File[]) {
     await writeFileToFS("/stats.db", await acceptedFiles[0].arrayBuffer());
-    const newData = await runAnalysis();
+    const newData = await runGameTickAnalysis('jerome-o');
     setData(newData);
-    console.log(data);
+    console.log(data); // 'data' isn't updated until the scope of onDrop is finished
   }
 
   async function useDefault() {
     await writeFileToFS(
       "/stats.db",
       await fetch("/databases/rlw_1.db").then((res) => res.arrayBuffer())
     );
-    const newData = await runAnalysis();
+    const newData = await runGameTickAnalysis('jerome-o');
+    console.log("Setting energy data")
     setData(newData);
-    console.log(data);
+    console.log("Energy data has been set:")
+    console.log(newData);
+    const tileData = await runTileDataAnalysis('jerome-o');
```

Comment:
This will only happen on the `useDefault` pathway and now the `onDrop` above - I reckon it should be refactored so that both `onDrop` and `useDefault` end up calling some ~run analysis~ kinda function

Reply:
@ksefonte

Reply:
`useDefault` and `onDrop` now call `processFile` which has the console logs - although I figure we could remove those after debugging?